chore: update browser example with stardust

vasco-santos opened the pull request:
This PR adds instructions on `libp2p-in-the-browser` example to create a setup using [libp2p-stardust](https://github.com/libp2p/libp2p-stardust) instead of `libp2p-webrtc` (or combining).

It was also used to test the correctness of the `libp2p-stardust` implementation

Needs:

- [x] [libp2p/js-libp2p-stardust#14](https://github.com/libp2p/js-libp2p-stardust/pull/14)
- [ ] final stardust release

vasco-santos commented:
> The reason for this is to avoid people needing to edit any code to see both of these working. 

We can have that once we setup infra for both signalling servers. Otherwise, they always will need to change the `peer-id` of the server. Unless we always use the same, which I was trying to avoid.

I will read better your remaining suggestions next week.

mkg20001 commented:
> We can have that once we setup infra for both signalling servers. Otherwise, they always will need to change the peer-id of the server. Unless we always use the same, which I was trying to avoid.

I think dialing without knowing ID is possible, so we can just dial ID-lessly. As discussed a MITM wouldn't have a tremendous impact aside from DOS since conns between peers are authed anyways.

jacobheun commented:
> Otherwise, they always will need to change the peer-id of the server. 

Ah, right, that makes sense.

jacobheun commented on examples/libp2p-in-the-browser/README.md:
````diff
@@ -15,20 +15,36 @@ npm install
 ```
 
-## Running the examples
+Despite being `libp2p-webrtc-star` being used, we have this example set up in a way that you can also use `libp2p-stardust` (you can remove `libp2p-webrtc-star` and just use it, but the more transports you support, the more likely you are to connect/discover other peers).
````
```suggestion
Although `libp2p-webrtc-star` is being used by default, we have this example set up in a way that you can also use `libp2p-stardust` (you can replace `libp2p-webrtc-star` with it, but it's recommended to use multiple transports if you want to increase the potential peers you can communicate with).
```

jacobheun commented on examples/libp2p-in-the-browser/index.js:
```diff
@@ -46,8 +47,12 @@ document.addEventListener('DOMContentLoaded', async () => {
+
+  const stardustServerId = 'QmQuyn6HEZrEC1Ex3SehY7f2sECrc7mM2KMAfTc2SDJD4X'
+  const stardustAddr = `/ip4/0.0.0.0/tcp/5892/ws/p2p-stardust/p2p/${stardustServerId}/p2p/${libp2p.peerInfo.id.toB58String()}`
+  libp2p.peerInfo.multiaddrs.add(stardustAddr)
```
Per the readme, startdust should be commented out and webrtc should be uncommented. Someone did the exercise ;)

vasco-santos replied:
ahah, my bad! I forgot to go back in time. Updating it

vasco-santos commented:
@jacobheun Addressed your review and also renamed the scripts to make them more understandable.

Regarding having both transports connected without user needing to change configuration, I agree with you that it would be the ideal situation. Once we have the servers deployed, I will change this example on that way, and I will also consider your thoughts on having a separation if we connect over the same protocol all the time.

vasco-santos commented:
@jacobheun @mkg20001 should we go forward with `stardust` release as both this PR and [ipfs/js-ipfs#2831](https://github.com/ipfs/js-ipfs/pull/2831) got an initial review? This will unblock me to ask libp2p infra team to get some stardust servers deployed

jacobheun commented on examples/libp2p-in-the-browser/package.json:
```diff
@@ -9,7 +9,8 @@
     "test": "echo \"Error: no test specified\" && exit 1",
     "start": "parcel index.html",
-    "server": "star-signal"
+    "webrtc-star": "star-signal",
```
This is another option for running the `webrtc-star` server and it will also match the logging mentioned in the readme.
```suggestion
    "webrtc-star": "webrtc-star",
```

jacobheun commented on examples/libp2p-in-the-browser/README.md:
```diff
@@ -15,20 +15,36 @@ npm install
+$ npm run stardust
+
+> libp2p-in-browser@1.0.0 server
+> stardust
```
logs `stardust-server`
```suggestion
> stardust-server
```

jacobheun commented on examples/libp2p-in-the-browser/README.md:
````diff
@@ -53,3 +69,37 @@ This will compile the code and start a server listening on port [http://localhos
+// libp2p.peerInfo.multiaddrs.add(stardustAddr)
+```
+
+You will just need to uncomment the lines above and replace the `stardustServerId` value for your server peerID. Moreover, if you want to only use `stardust`, you can comment the `webrtcAddr` definition above `stardust`.
````
```suggestion
You will just need to uncomment the lines above and replace the `stardustServerId` value for your server peerID. Moreover, if you want to only use `stardust`, you can comment the `webrtcAddr` definition above `stardust`. Once you save the file parcel should automatically reload the browser tabs.
```

jacobheun commented on examples/libp2p-in-the-browser/README.md:
```diff
@@ -53,3 +69,37 @@ This will compile the code and start a server listening on port [http://localhos
+
+Now, if you open a second browser tab to `http://localhost:1234`, you should discover your node from the previous tab. This is due to the fact that the `libp2p-stardust` transport also acts as a Peer Discovery interface. Your node will be notified of any peer that connects to the same signaling server you are connected to. Once libp2p discovers this new peer, it will attempt to establish a direct ...
+
+**Note**: In the example we assign libp2p to `window.libp2p`, in case you would like to play around with the API directly in the browser. You can of course make changes to `index.js` and Parcel will automatically rebuild and reload the browser tabs.
```
This is all redundany, I'd just remove it.
```suggestion
```

vasco-santos replied:
From a user perspective it might make things confusing though. If I come to this example and I am only interested in `stardust`, I should not need to read the `webrtc` part to properly run the `stardust`.

Should we add a notice to the users to check the running instructions from `webrtc`?


##  Setup stardust

## Run the example

jacobheun replied:
Or create a section on running the parcel web services and create a link to it.

vasco-santos commented:
@mkg20001 can you release `libp2p-stardust@0.2.0`?  We can iterated on later releases on the open issues

jacobheun commented:
Closing as we're not moving forward with this.
